src/main/java/CancelApplication/CancelApplicationInteractor.java

```diff
@@ -3,17 +3,32 @@
 import Entities.Group;
 import Entities.User;
 
+/**
+ * The cancelApplication use case.
+ * The user can only cancel applications for groups that they have previously applied to,
+ * where the user has not already been rejected or approved.
+ */
+
 public class CancelApplicationInteractor implements CancelApplicationInputBoundary {
 
     final CancelApplicationDsGateway dsGateway;
     final CancelApplicationOutputBoundary outputBoundary;
 
+    /**
+     * The interactor for the cancelApplication use case.
+     * @param dsGateway the data access interface
+     * @param outputBoundary the output boundary implemented by the cancelApplicationPresenter
+     */
     public CancelApplicationInteractor(CancelApplicationDsGateway dsGateway,
                                        CancelApplicationOutputBoundary outputBoundary) {
         this.dsGateway = dsGateway;
         this.outputBoundary = outputBoundary;
     }
 
+    /**
+     * @param requestModel the requestModel for the cancelApplication use case
+     * @return a successView or failView depending on whether the user has cancelled their group application
+     */
     @Override
     public CancelApplicationResponseModel cancelApplication(CancelApplicationRequestModel requestModel) {
 
```

Comment:
See, I was unsure about that. Originally I had that check, as you said, because the username might be incorrect which would raise a nullpointer exception since I can't get the user from the repo.
But now we have a current user. It's initialized as soon as you log in (from my understanding after talking to Leo) and so his usecase sets the current user, which is the one currently logged in, which means the current user HAS to exist because they've already logged in. 
So I don't know if having this check is redundant. I'll talk to Sharon and Ipek about this -- I think they have the same check.

Reply:
I agree with Aarya. It does seem rather redundant. However, for us not to need that check, we would have to be 100% certain that we are grabbing the username correctly, and/or the user creation use case functions correctly. If we work under that assumption (I think we should otherwise we have a faulty app which is a problem in it of itself), then I do not believe it needs to be counted. Of course, check with Ipek and Sharon as they are doing something very similar.